docs/DeveloperGuide.md

```diff
@@ -591,42 +633,330 @@ testers are expected to do more *exploratory* testing.
+
+   2. **Test Case**: `add-leave id/x sd/01-01-2022 ed/02-01-2022`<br>
+      **Expected Outcome**: No changes occured. An error message appears, saying that overlapping leaves are not allowed.
+
+   3. **Test Case**: `add-leave id/x sd/31-04-2022 ed/01-05-2022`<br>
+      **Expected Outcome**: No changes occured. An error message appears, saying that there were invalid date inputs.
+
+   4. **Test Case**: `add-leave id/x sd/01-05-2022 ed/15-05-2022`<br>
+      **Expected Outcome**: No changes occured. An error message appears, saying that the employee does not have enough leaves left.
+      This is assuming that the total leaves is 14, the default value.
+      Else, you can adjust the end date in the example provided to achieve a duration longer than the available leaves.
+
+### Deleting employee leaves
+
+For the following tests, ensure that there is at least 1 employee, with the default total leaves of 14 days.
+If there are more than one, any of these employees can be used, and can be used for different test cases as well.
+In each of the test cases, we will refer to this employee's ID as `x`, for generality.
+
+In addition, prior to running each test case, ensure that the employee has at least one leave period.
+
+**Note**: Changes are mostly visible only when viewing the employee profile.
+Therefore, we recommend that when performing a test case for an employee (with ID `x`), you first view the profile to easily observe the outcome.
+
+1. Deleting leave period for an employee
+
+   1. **Test Case**: `delete-leave id/x i/1`<br>
+      **Expected Outcome**: Total leaves remain the same.
+      Leaves left increases by the duration of the corresponding leave period.
+      The corresponding row in the leaves table is deleted.
+
+   2. **Test Case**: `delete-leave id/x i/y+1`, where `y` is the number of leave periods the employee has<br>
+      **Expected Outcome**: No changes occured. An error message appears, saying that the index provided is invalid.
+
+   3. **Test Case**: `delete-leave id/x i/0`<br>
+      **Expected Outcome**: No changes occured. An error message appears, providing the constraints for numbers.
 
-1. _{ more test cases …​ }_
+### Rating employee performance
 
-### Deleting a person
+For the following tests, ensure that there is at least 1 employee, who has never been rated prior.
+If there are more than one, any of these employees can be used, and can be used for different test cases as well.
+In each of the test cases, we will refer to this employee's ID as `x`, for generality.
 
-1. Deleting a person while all persons are being shown
+**Note**: Changes are mostly visible only when viewing the employee profile.
+Therefore, we recommend that when performing a test case for an employee (with ID `x`), you first view the profile to easily observe the outcome.
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+1. Rating an employee's performance
```

Comment:
Should include the scenario where user rating is out of the acceptable range, example r/6

Reply:
Okay will add.